src/tests/ftest/util/ior_utils.py

```diff
@@ -1,110 +1,267 @@
+
+        """
+        for name, ior_param in self.__dict__.items():
+            ior_param.set_yaml_value(name, test, path)
+
+    def get_launch_command(self, basepath, processes, hostfile):
+        """Get the process launch command used to run IOR.
+
+        Args:
+            basepath (str): DAOS base path
+            processes (int): number of host processes
+            hostfile (str): file defining host names and slots
+
+        Raises:
+            IorFailed: if an error occured building the IOR command
+
+        """
+        with open(os.path.join(basepath, ".build_vars.json")) as afile:
+            build_paths = json.load(afile)
+        attach_info_path = os.path.join(basepath, "install/tmp")
+
+        # # Verify required params are defined
+        # for name in ("daos_pool", "daos_cont", "daos_svcl"):
+        #     value = getattr(self, name)
+        #     if str(value) == "":
+        #         raise IorFailed("Missing required {} param".format(name))
+
+        if self.api.value == "MPIIO":
+            env = {
+                "CRT_ATTACH_INFO_PATH": attach_info_path,
+                "DAOS_POOL": str(self.daos_pool),
+                "MPI_LIB": "",
+                "DAOS_SVCL": str(self.daos_svcl),
+                "DAOS_SINGLETON_CLI": 1,
+                "FI_PSM2_DISCONNECT": 1,
+            }
+            export_cmd = [
+                "export {}={}".format(key, val) for key, val in env.items()]
+            mpirun_cmd = [
+                os.path.join(build_paths["OMPI_PREFIX"], "bin/mpirun"),
+                "-np {}".format(processes),
+                "--hostfile {}".format(hostfile),
+            ]
+            command = " ".join(mpirun_cmd + [self.__str__()])
+            command = "; ".join(export_cmd + command)
+
+        elif self.api.value == "DAOS":
+            orterun_cmd = [
+                os.path.join(build_paths["OMPI_PREFIX"], "bin/orterun"),
+                "-np {}".format(processes),
+                "--hostfile {}".format(hostfile),
+                "--map-by node",
+                "-x DAOS_SINGLETON_CLI=1",
+                "-x CRT_ATTACH_INFO_PATH={}".format(attach_info_path),
+            ]
+            command = " ".join(orterun_cmd + [self.__str__()])
+
+        else:
+            raise IorFailed("Unsupported IOR API: {}".format(self.api))
+
+        return command
+
+    def run(self, basepath, processes, hostfile, display_output=True):
+        """Run the IOR command.
+
+        Args:
+            basepath (str): DAOS base path
+            processes (int): number of host processes
+            hostfile (str): file defining host names and slots
+            display_output (bool, optional): print IOR output to the console.
+                Defaults to True.
+
+        Raises:
+            IorFailed: if an error occured runnig the IOR command
+
+        """
+        command = self.get_launch_command(basepath, processes, hostfile)
```

Comment:
This new argument then needs to be passed to `get_launch_command`:

`command = self.get_launch_command(basepath, processes, hostfile, run_path)`